### ISSUE TITLE
improve auto-sort

### ISSUE THREAD

**Issue by swharden** [swharden/LJPcalc#19]
Notes from Dr. Brogioli, 

_The image that I have in mind is that "x" and "last" should have different sign (if "x" is a cation, "last" is the most common anion). However, I found no practical case in which this matters._

_You [currently] select "last" as the ion with largest cL, with the risk that c0 is very small. I do not know if this is a problem, maybe not. However, I was thinking if it is not better to first select "x", then select "last" among the remaining ions, as the one having the largest c0+cL._

The auto-sort logic lives here:
https://github.com/swharden/LJPcalc/blob/5ffd25784dabaf9acbd34eaa6f4da44e0ed9cb0e/src/LJPmath/Calculate.cs#L10-L43

**Comment by swharden** [comment]
![image](https://user-images.githubusercontent.com/4165489/82959225-3660c600-9f85-11ea-8962-18ed63ca9a52.png)

![image](https://user-images.githubusercontent.com/4165489/82959242-4082c480-9f85-11ea-8d77-a22ef795550e.png)

this error was due to localization settings and parsing of periods vs commas so mobilities are crazy numbers

**Comment by swharden** [comment]
Restating Dr. Brogioli, the goal of the proposed automatic ion sorting strategy is to have `last` and `x` (second-to-last) have charges with different signs, and ensure `x` has a large `c0`. A constraint is that `x` cannot have equal `c0` and `cL` as this will crash the phi solver.

I tried this, but it makes the test ion sequences fail (calculated LJPs are accurate, but changing the sort rules causes the phi solver to fail for some sequences). This is one of the finicky ones that fails easily:
```cs
var ionSet = new List<Ion>()
	{
		new Ion("Zn", 9, .0284),
		new Ion("K", 0, 3),
		new Ion("Cl", 18, 3.062),
		new Ion("Mg", 5, 3),
		new Ion("Ag", 1, 1),
	};
```

After working on this for a while I decided not to change the sorting algorithm since the current one is supported by so many tests and I haven't been able to find an example ion set that makes it fail.
